sdks/wasm/library_mono.js

```diff
@@ -0,0 +1,24 @@
+
+var MonoSupportLib = {
+	$MONO__postset: 'Module["pump_message"] = MONO.pump_message',
+	$MONO: {
+		pump_count: 0,
+		pump_message: function () {
+			while (MONO.pump_count > 0) {
+				--MONO.pump_count;
+				Module.cwrap ("mono_gc_pump_callback") ();
+			}
+		}
+	},
+
+	request_gc_cycle: function () {
+		++MONO.pump_count;
+		if (Module.ENVIRONMENT_IS_WEB) {
```

Comment:
I don't think this is correct. In `mono.js`, `ENVIRONMENT_IS_WEB` is a local variable, and is not exposed as a property of `Module`, so this check always evaluates as false and GC never runs. I think you need something like the following instead:

```
if (Module['ENVIRONMENT'] === 'WEB' || Module['ENVIRONMENT'] === 'WORKER')
```

... which not only fixes the `WEB` case but also allows GC in the `WORKER` case.